python-spec/src/somacore/data.py

```diff
@@ -88,8 +88,9 @@ def shape(self) -> Tuple[int, ...]:
         raise NotImplementedError()
 
     @property
-    def ndims(self) -> int:
-        """The Number of Dimensions in this array."""
+    @abc.abstractmethod
```

Comment:
Since this is the implementation of ndim, this should not be marked as abstract.